libraries/src/Mail/MailTemplate.php

```diff
@@ -485,29 +486,20 @@ public static function deleteTemplate($key)
 	 *
 	 * @since   4.0.0
 	 */
-	protected function setAttachmentName($file, $name)
+	protected function getAttachmentName($file, $name)
 	{
-		// If no name is given use the basename of the file to be attached
+		// If no name is given, do not process it further
 		if (!trim($name))
 		{
-			return basename($file);
+			return '';
 		}
 
-		// Get file extension of the file to be attached
-		$extensionFile = pathinfo($file, PATHINFO_EXTENSION);
-
-		// If the file has no extension there is nothing to do with the name
-		if (!$extensionFile)
-		{
-			return $name;
-		}
-
-		// Make sure to attach the file with the same extension as the attached file has (case-insensitive)
-		if (strtolower(pathinfo($name, PATHINFO_EXTENSION)) !== strtolower($extensionFile))
+		// The given name use a different file extension, something could not be right, ignore it
+		if (File::getExt($file) !== File::getExt($name))
 		{
-			return $name . '.' . $extensionFile;
+			return '';
```

Comment:
I'd prefer to fix the extension of the attachment name $name and not completely fall back to the real file name here. Maybe it can be made nicer than I did it, e.g. removing any extension from the attachment name before appending the right extension.

Or maybe even better, not allow dots in the attachment name so we have no extension, and always append the one of the real file.

Maybe, if we can't use makeSafe, we can use client-side html5 filtering (property "pattern" of the text field) to disallow certain characters, including dots, slashes, back-slashes. But that seems to be tricky to get it working. I've tried and it seems you can't do everything which you can do with "Normal "regular expressions.

Reply:
If we really want to fix it, the only thing we should do if people do not enter file extension, we append extension of the original file to that name. If someone enters test.exe, we can just ignore it instead of fixing it to test.exe.jpg. We could not fix all the stupid.

Reply:
That could be the way, see my comments above on how we can do it.